File: sdks/csharp/src/NovaBilling/Wallets/Requests/TopUpWalletDto.cs
using System.Text.Json.Serialization;
using NovaBilling.Core;

namespace NovaBilling;

[Serializable]
public record TopUpWalletDto
{
    [JsonPropertyName("walletId")]
    public required string WalletId { get; set; }

    /// <summary>
    /// Paid credits to purchase
    /// </summary>
    [JsonPropertyName("paidCredits")]
    public double? PaidCredits { get; set; }

    /// <summary>
    /// Free credits to grant
    /// </summary>
    [JsonPropertyName("grantedCredits")]
    public double? GrantedCredits { get; set; }

    /// <summary>
    /// Credits to void
    /// </summary>
    [JsonPropertyName("voidedCredits")]
    public double? VoidedCredits { get; set; }

    [JsonPropertyName("metadata")]
    public Dictionary<string, object?>? Metadata { get; set; }

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}

File: sdks/php/src/Wallets/Requests/TopUpWalletDto.php
<?php

namespace NovaBilling\Wallets\Requests;

use NovaBilling\Core\Json\JsonSerializableType;
use NovaBilling\Core\Json\JsonProperty;
use NovaBilling\Core\Types\ArrayType;

class TopUpWalletDto extends JsonSerializableType
{
    /**
     * @var string $walletId
     */
    #[JsonProperty('walletId')]
    public string $walletId;

    /**
     * @var ?float $paidCredits Paid credits to purchase
     */
    #[JsonProperty('paidCredits')]
    public ?float $paidCredits;

    /**
     * @var ?float $grantedCredits Free credits to grant
     */
    #[JsonProperty('grantedCredits')]
    public ?float $grantedCredits;

    /**
     * @var ?float $voidedCredits Credits to void
     */
    #[JsonProperty('voidedCredits')]
    public ?float $voidedCredits;

    /**
     * @var ?array<string, mixed> $metadata
     */
    #[JsonProperty('metadata'), ArrayType(['string' => 'mixed'])]
    public ?array $metadata;

    /**
     * @param array{
     *   walletId: string,
     *   paidCredits?: ?float,
     *   grantedCredits?: ?float,
     *   voidedCredits?: ?float,
     *   metadata?: ?array<string, mixed>,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->walletId = $values['walletId'];
        $this->paidCredits = $values['paidCredits'] ?? null;
        $this->grantedCredits = $values['grantedCredits'] ?? null;
        $this->voidedCredits = $values['voidedCredits'] ?? null;
        $this->metadata = $values['metadata'] ?? null;
    }
}
